cmake/download-external-project.cmake

```diff
@@ -1,17 +1,21 @@
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+option(HIT_REBUILD_DEPS "Rebuild dependendies" OFF)
```

Comment:
What is the behavior for someone who freshly clones the repo? Won't they get errors if they forget to turn this flag ON because the dependencies aren't getting built?

Reply:
Nope. This will always build the dependencies if cmake things that it is necessary (which includes "I have the source but no binaries".) This just removes unneeded *re*builds.